### PR TITLE
chore: migrate to main branch

### LINKED ISSUE #216: Migrate to main branch
Contributing to googleapis/google-cloud-python#10579, switching the default branch to main branch. Also replacing all occurrences of master to main.

### PR DESCRIPTION
**Pull request by dandhlee**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-datastore/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #216  🦕

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -13,7 +13,7 @@ all other queries.
 - `Product Documentation`_
 
 .. |GA| image:: https://img.shields.io/badge/support-GA-gold.svg
-   :target: https://github.com/googleapis/google-cloud-python/blob/master/README.rst#general-availability
+   :target: https://github.com/googleapis/google-cloud-python/blob/main/README.rst#general-availability
```
Comment (parthea) [inline README.rst:16]: I get 404 with the new link

**Review by parthea — COMMENTED** [review]
One observation, otherwise LGTM

File: README.rst (hunk 1 above)
Comment (dandhlee) [reply README.rst]: Done.

**Comment by dandhlee** [comment]
Please take a look again!

**Review by parthea — APPROVED** [review]

**Pull request merged**